### DISCUSSION TITLE
Switch between data files to browse charts

### DISCUSSION THREAD

**Discussion by vbozhkov** [highfestiva/finplot#523] [Ideas]
Hi! Amazing work you have done! I'm tired of paying for subscriptions that do nothing when I'm already paying for market data from 3 places... Is it possible to switch between files with arrow key or with a button or something. lets say i've saved all the tickers in a folder with AAPL.csv, MSFT.csv, NVDA.csv etc. and i want to browse the different tickers without relaunching, is this possible?

**Comment by vbozhkov** [comment]
I have successfully implemented this with pyplot, but pyplot sucks, would be pretty awesome if this could be done with finplot 
example code:

# Function to load data from CSV
def load_data(file_path):
    if os.path.exists(file_path):
        df = pd.read_csv(file_path)
        df['date'] = pd.to_datetime(df['date'])
        df.set_index('date', inplace=True)
        df['MA10'] = df['close'].rolling(window=10).mean()
        df['MA20'] = df['close'].rolling(window=20).mean()
        return df
    else:
        print(f"No data file found at {file_path}.")
        return None

# Function to plot data
def plot_data(df, file_name):
    plt.clf()
    plt.title(file_name)
    plt.plot(df.index, df['close'], label='Close')
    plt.plot(df.index, df['MA10'], label='MA10')
    plt.plot(df.index, df['MA20'], label='MA20')
    plt.legend()
    plt.draw()

# Key event handling function
def on_key(event):
    global current_index
    if event.key == ' ':
        current_index = (current_index + 1) % len(csv_files)
        df = load_data(csv_files[current_index])
        if df is not None:
            plot_data(df, os.path.basename(csv_files[current_index]))

# Directory containing the CSV files
folder = "Charts"
csv_files = [os.path.join(folder, file) for file in os.listdir(folder) if file.endswith('.csv')]

if not csv_files:
    print("No CSV files found in the specified folder.")
else:
    # Set up interactive plot
    current_index = 0
    fig, ax = plt.subplots()
    fig.canvas.mpl_connect('key_press_event', on_key)

    # Initial plot
    df = load_data(csv_files[current_index])
    if df is not None:
        plot_data(df, os.path.basename(csv_files[current_index]))

    plt.show()

**Comment by highfestiva** [comment]
Sure you can! The complicated.py example shows how to clear the plot and create new ones when you select some new asset. Similar things can be seen in other examples. Basically you just do this:

```python3
def change_asset():
    ax.reset() # remove previous plots
    fplt.plot(df['close'])
```
